### ISSUE TITLE
navigator.webdriver still can be detected

### ISSUE THREAD

**Issue by felyxjet** [berstend/puppeteer-extra#222]
```
const puppeteer = require('puppeteer-extra')
const StealthPlugin = require('puppeteer-extra-plugin-stealth')
puppeteer.use(StealthPlugin())

puppeteer.launch({ headless: true }).then(async browser => {
    const page = await browser.newPage()
    page.on('console', msg => console.log(msg.text()))
    await page.goto('https://bot.sannysoft.com')
    await page.evaluate(async () => {
        console.log('webdriver' in navigator ? 'detected': 'not detected')
        console.log(navigator.webdriver === undefined ? 'not detected': 'detected')
        // but!
        console.log(Object.getOwnPropertyDescriptor(navigator.__proto__, 'webdriver') === undefined ? 'not detected': 'detected once')
        for (let prop in navigator) {
            if (prop === 'webdriver') {
                console.log('detected twice')
            }
        }
    });

    await browser.close()
})
```

```
const newProto = navigator.__proto__;
delete newProto.webdriver;
navigator.__proto__ = newProto
```
was more correct way to remove `webdriver` from `navigator`.

**Comment by Torpedo99** [comment]
If you're referring to #207 then please do make a PR for the changes and link them if you have the time. This new method doesn't seem to be used much but just to make sure. Thanks

**Comment by felyxjet** [comment]
No, i'm not referring to that. I'm pointing out that removing all traces of `webdriver` by proxying `navigator` is impossible.

**Comment by Niek** [comment]
If you want to delete it at every page you need to execute it in [page.evaluateOnNewDocument](https://github.com/puppeteer/puppeteer/blob/v3.0.4/docs/api.md#pageevaluateonnewdocumentpagefunction-args).

**Comment by momala454** [comment]
> 
> 
> This is the right way to delete it:
> 
> ```js
> delete navigator.__proto__.webdriver;
> ```

as `__proto__` is deprecated, maybe use Object.getPrototypeOf() ?

**Comment by sam4652** [comment]
> If you want to delete it at every page you need to execute it in [page.evaluateOnNewDocument](https://github.com/puppeteer/puppeteer/blob/v3.0.4/docs/api.md#pageevaluateonnewdocumentpagefunction-args).

Thanks, now it is detected befor but not after goto, it's the most important i think. 
I've turned off puppeteer-extra-plugin-stealth/evasions/navigator.webdriver

**Comment by sam4652** [comment]
> > This is the right way to delete it:
> > ```js
> > delete navigator.__proto__.webdriver;
> > ```
> 
> as `__proto__` is deprecated, maybe use Object.getPrototypeOf() ?

Deprecated means non-functional on some browsers ? Me on the browser I use the console.log are ok, is it good in my case to use _proto_  ?

**Comment by momala454** [comment]
deprecated means it could be removed at any time, or could not work properly

**Comment by Bllacky** [comment]
> 
> 
> deprecated means it could be removed at any time, or could not work properly

Exactly! Therefore delete navigator.__proto__.webdriver is not a reliable solution. We need to implement the same thing but without using _proto_

**Comment by Niek** [comment]
There is no way ``__proto__`` will get deleted from Chrome anytime soon, it will break too many sites. But in case it will, this works as well:
```javascript
delete Object.getPrototypeOf(navigator).webdriver;
```

**Comment by sam4652** [comment]
It works well thank you. However before the goto webdriver is always true and just after the goto if the site is fast it can perhaps see that it is true before we even delete it. What would the solution be to delete it before goto too ?

**Comment by sachikocat** [comment]
Hi, I need some help. I already follow the suggestion to add `delete Object.getPrototypeOf(navigator).webdriver;`. So my code will be like this
`(async function main() {
	// puppeteer-extra is a drop-in replacement for puppeteer,
	// it augments the installed puppeteer with plugin functionality
	const puppeteer = require('puppeteer-extra');
	 
	// add stealth plugin and use defaults (all evasion techniques)
	const stealth = require('puppeteer-extra-plugin-stealth')();
	stealth.onBrowser = () => {};
	puppeteer.use(stealth);
	 
	// puppeteer usage as normal
	puppeteer.launch({ headless: false }).then(async browser => {
		const page = await browser.newPage()
	    page.on('console', msg => console.log(msg.text()))
	    await page.goto('https://ib.bri.co.id/ib-bri/')
	    await page.evaluate(async () => {
	    	delete Object.getPrototypeOf(navigator).webdriver;
	        console.log('webdriver' in navigator ? 'detected': 'not detected')
	        console.log(navigator.webdriver === undefined ? 'not detected': 'detected')
	        // but!
	        console.log(Object.getOwnPropertyDescriptor(navigator.__proto__, 'webdriver') === undefined ? 'not detected': 'detected once')
	        for (let prop in navigator) {
	            if (prop === 'webdriver') {
	                console.log('detected twice')
	            }
	        }
	    });
	    await page.waitFor(5000)
	    await browser.close()
	})
})();`

I got this on my browser.
![image](https://user-images.githubusercontent.com/67307841/85367011-a184c480-b552-11ea-9c1f-edb95949d33b.png)

And in my console log show this message
![CMD BNI](https://user-images.githubusercontent.com/67307841/85367051-b3fefe00-b552-11ea-8913-95165641da08.PNG)

Any advice for me? Thank you

**Comment by felyxjet** [comment]
Remove driver in `evaluateOnNewDocument`. It makes no sense to do that after page is loaded.

**Comment by berstend** [comment]
Fixed in #244, published in `puppeteer-extra-plugin-stealth@2.4.13`. :-)

**Comment by codeex** [comment]
very good！ `delete navigator.__proto__.webdriver;`
